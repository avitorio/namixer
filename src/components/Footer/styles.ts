import styled, { css } from 'styled-components'

export const Wrapper = styled.footer`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    margin-top: ${theme.spacings.medium};
    color: ${theme.colors.white};
    text-align: center;
    line-height: 2.4rem;

    a {
      color: ${theme.colors.white};
    }
  `}
`

export const Copyright = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.xsmall};
    margin-top: ${theme.spacings.large};
    margin-bottom: ${theme.spacings.medium};
    text-align: center;
  `}
`
