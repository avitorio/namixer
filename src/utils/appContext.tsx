import {
  Dispatch,
  SetStateAction,
  useState,
  useContext,
  createContext,
  ReactNode
} from 'react'

export interface IProviderProps {
  children?: ReactNode
}

type AppContextState = { domain: string }

type AppContextValue = {
  state: AppContextState
  setState: Dispatch<SetStateAction<AppContextState>>
}

export const AppContext = createContext<AppContextValue | undefined>(undefined)

export const AppProvider = (props: IProviderProps) => {
  const [state, setState] = useState({ domain: 'YourDomain.com' })

  return (
    <AppContext.Provider value={{ state, setState }}>
      {props.children}
    </AppContext.Provider>
  )
}

export function useAppContext() {
  const ctxValue = useContext(AppContext)
  if (ctxValue === undefined)
    throw new Error(
      'Expected an AppProvider somewhere in the react tree to set context value'
    )
  return ctxValue // now has type AppContextValue
  // or even provide domain methods for better encapsulation
}
