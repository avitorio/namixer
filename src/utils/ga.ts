// https://developers.google.com/analytics/devguides/collection/gtagjs/pages
export const pageview = (url: URL): void => {
  window.gtag('config', `${process.env.NEXT_PUBLIC_GA_TRACKING}`, {
    page_path: url
  })
}

type EventParams = { [key: string]: string | number }

type GTagEvent = {
  event: string
  params: EventParams
}

// https://developers.google.com/analytics/devguides/collection/gtagjs/events
export const event = ({ event, params }: GTagEvent): void => {
  window.gtag('event', event, {
    ...params
  })
}
