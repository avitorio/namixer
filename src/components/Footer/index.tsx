import Logo from 'components/Logo'
import * as S from './styles'

const currentYear = new Date().getFullYear()

const Footer = () => (
  <S.Wrapper>
    <Logo color="black" />
    <S.Content>
      <p>
        Built by Andre Vitorio. Follow me on{' '}
        <a href="https://twitter.com/andrevitorio">Twitter</a> and{' '}
        <a href="https://linkedin.com/in/andrevitorio">LinkedIn</a>.
      </p>
    </S.Content>

    <S.Copyright>Namixer {currentYear} Â© All rights reserved.</S.Copyright>
  </S.Wrapper>
)

export default Footer
