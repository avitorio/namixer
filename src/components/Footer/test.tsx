import { render } from 'utils/test-utils'

import Footer from '.'

describe('<Footer />', () => {
  it('should render the footer.', () => {
    const { container } = render(<Footer />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
