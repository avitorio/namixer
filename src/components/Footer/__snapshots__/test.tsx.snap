// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the footer. 1`] = `
.c1 {
  color: #FAFAFA;
  width: 15rem;
  height: 3.3rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 3.2rem;
  color: #FAFAFA;
  text-align: center;
  line-height: 2.4rem;
}

.c2 a {
  color: #FAFAFA;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.2rem;
  margin-top: 4.0rem;
  margin-bottom: 3.2rem;
  text-align: center;
}

<footer
  class="c0"
>
  <div
    class="c1"
    color="white"
    id=""
  >
    <svg
      aria-label="Namixer"
      fill="none"
      role="img"
      viewBox="0 0 374 92"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="text"
        d="M121.646 71V26.012H132.065L143.795 48.368L148.211 58.304H148.487C148.303 55.912 148.05 53.244 147.728 50.3C147.406 47.356 147.245 44.55 147.245 41.882V26.012H156.905V71H146.486L134.756 48.575L130.34 38.777H130.064C130.294 41.261 130.547 43.929 130.823 46.781C131.145 49.633 131.306 52.393 131.306 55.061V71H121.646ZM175.33 71.828C172.202 71.828 169.718 70.839 167.878 68.861C166.038 66.837 165.118 64.376 165.118 61.478C165.118 57.89 166.636 55.084 169.672 53.06C172.708 51.036 177.607 49.679 184.369 48.989C184.277 47.471 183.817 46.275 182.989 45.401C182.207 44.481 180.873 44.021 178.987 44.021C177.561 44.021 176.112 44.297 174.64 44.849C173.168 45.401 171.604 46.16 169.948 47.126L166.291 40.433C168.453 39.099 170.753 38.018 173.191 37.19C175.675 36.362 178.251 35.948 180.919 35.948C185.289 35.948 188.647 37.213 190.993 39.743C193.339 42.273 194.512 46.16 194.512 51.404V71H186.232L185.542 67.481H185.266C183.84 68.769 182.299 69.827 180.643 70.655C179.033 71.437 177.262 71.828 175.33 71.828ZM178.78 63.962C179.93 63.962 180.919 63.709 181.747 63.203C182.621 62.651 183.495 61.938 184.369 61.064V55.061C180.781 55.521 178.297 56.257 176.917 57.269C175.537 58.235 174.847 59.385 174.847 60.719C174.847 61.823 175.192 62.651 175.882 63.203C176.618 63.709 177.584 63.962 178.78 63.962ZM203.092 71V36.776H211.372L212.062 41.192H212.338C213.764 39.766 215.282 38.547 216.892 37.535C218.502 36.477 220.434 35.948 222.688 35.948C225.126 35.948 227.081 36.454 228.553 37.466C230.071 38.432 231.267 39.835 232.141 41.675C233.659 40.111 235.269 38.777 236.971 37.673C238.673 36.523 240.651 35.948 242.905 35.948C246.585 35.948 249.276 37.19 250.978 39.674C252.726 42.112 253.6 45.47 253.6 49.748V71H243.457V51.059C243.457 48.575 243.112 46.873 242.422 45.953C241.778 45.033 240.72 44.573 239.248 44.573C237.546 44.573 235.591 45.677 233.383 47.885V71H223.24V51.059C223.24 48.575 222.895 46.873 222.205 45.953C221.561 45.033 220.503 44.573 219.031 44.573C217.329 44.573 215.397 45.677 213.235 47.885V71H203.092ZM262.254 71V36.776H272.397V71H262.254ZM267.291 31.739C265.589 31.739 264.209 31.256 263.151 30.29C262.093 29.324 261.564 28.036 261.564 26.426C261.564 24.816 262.093 23.528 263.151 22.562C264.209 21.596 265.589 21.113 267.291 21.113C268.993 21.113 270.373 21.596 271.431 22.562C272.489 23.528 273.018 24.816 273.018 26.426C273.018 28.036 272.489 29.324 271.431 30.29C270.373 31.256 268.993 31.739 267.291 31.739ZM277.804 71L288.292 53.198L278.425 36.776H289.327L292.363 42.296C292.823 43.262 293.306 44.251 293.812 45.263C294.318 46.229 294.801 47.195 295.261 48.161H295.537C295.905 47.195 296.296 46.229 296.71 45.263C297.124 44.251 297.515 43.262 297.883 42.296L300.229 36.776H310.717L300.781 54.371L311.338 71H300.436L297.124 65.342C296.572 64.376 296.02 63.387 295.468 62.375C294.962 61.363 294.433 60.374 293.881 59.408H293.605C293.191 60.374 292.754 61.363 292.294 62.375C291.834 63.341 291.397 64.33 290.983 65.342L288.292 71H277.804ZM330.265 71.828C326.999 71.828 324.055 71.115 321.433 69.689C318.811 68.263 316.741 66.216 315.223 63.548C313.705 60.88 312.946 57.66 312.946 53.888C312.946 50.162 313.705 46.965 315.223 44.297C316.787 41.629 318.811 39.582 321.295 38.156C323.779 36.684 326.378 35.948 329.092 35.948C332.358 35.948 335.049 36.684 337.165 38.156C339.327 39.582 340.937 41.537 341.995 44.021C343.099 46.459 343.651 49.242 343.651 52.37C343.651 53.244 343.605 54.118 343.513 54.992C343.421 55.82 343.329 56.441 343.237 56.855H322.744C323.204 59.339 324.239 61.179 325.849 62.375C327.459 63.525 329.391 64.1 331.645 64.1C334.083 64.1 336.544 63.341 339.028 61.823L342.409 67.964C340.661 69.16 338.706 70.103 336.544 70.793C334.382 71.483 332.289 71.828 330.265 71.828ZM322.675 50.162H335.026C335.026 48.276 334.566 46.735 333.646 45.539C332.772 44.297 331.323 43.676 329.299 43.676C327.735 43.676 326.332 44.228 325.09 45.332C323.848 46.39 323.043 48 322.675 50.162ZM350.66 71V36.776H358.94L359.63 42.779H359.906C361.148 40.479 362.643 38.777 364.391 37.673C366.139 36.523 367.887 35.948 369.635 35.948C370.601 35.948 371.406 36.017 372.05 36.155C372.694 36.247 373.246 36.408 373.706 36.638L372.05 45.401C371.452 45.217 370.854 45.079 370.256 44.987C369.704 44.895 369.06 44.849 368.324 44.849C367.036 44.849 365.679 45.332 364.253 46.298C362.873 47.218 361.723 48.828 360.803 51.128V71H350.66Z"
        fill="white"
      />
      <rect
        fill="#E65A6D"
        height="87"
        rx="7.5"
        stroke="black"
        stroke-width="5"
        width="87"
        x="2.5"
        y="2.5"
      />
      <path
        d="M22 19V72.5"
        stroke="black"
        stroke-linecap="round"
        stroke-width="6"
      />
      <path
        d="M46 19V72.5"
        stroke="black"
        stroke-linecap="round"
        stroke-width="6"
      />
      <path
        d="M46 19V72.5"
        stroke="black"
        stroke-linecap="round"
        stroke-width="6"
      />
      <path
        d="M70 19V72.5"
        stroke="black"
        stroke-linecap="round"
        stroke-width="6"
      />
      <circle
        cx="22"
        cy="23"
        fill="#D3FAC6"
        r="8"
        stroke="black"
        stroke-width="4"
      />
      <circle
        cx="46"
        cy="46"
        fill="#D3FAC6"
        r="8"
        stroke="black"
        stroke-width="4"
      />
      <circle
        cx="70"
        cy="23"
        fill="#D3FAC6"
        r="8"
        stroke="black"
        stroke-width="4"
      />
    </svg>
  </div>
  <div
    class="c2"
  >
    <p>
      Built by Andre Vitorio.
      <br />
       Follow me on
       
      <a
        href="https://twitter.com/andrevitorio"
        rel="noreferrer"
        target="_blank"
      >
        Twitter
      </a>
       
      and
       
      <a
        href="https://linkedin.com/in/andrevitorio"
        rel="noreferrer"
        target="_blank"
      >
        LinkedIn
      </a>
      .
    </p>
  </div>
  <div
    class="c3"
  >
    Namixer 
    2022
     Â© All rights reserved.
  </div>
</footer>
`
